lib/src/specs/type_record.dart

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
```

Comment:
[nit] Why `type_record.dart` instead of `record_type.dart`?

Reply:
I was following the `type_function.dart` example, but I don't have a strong opinion here.